Addon (plugin) section added

aborovsky opened the pull request:
Please, take a look at this.

gabriel-detassigny commented on app/scripts/services/addon.js:
```diff
@@ -0,0 +1,109 @@
+        //KodiWS.send('VideoLibrary.GetSeasons', { properties: ['season', 'episode', 'watchedepisodes'], addonid: parseInt(addonid) }).then(function(data) {
+        //  addon.seasons = data.seasons;
+        //  deferred.resolve(addon);
+        //});
```
Can you remove that comment? I suppose we don't need it

gabriel-detassigny commented on app/views/addons/addons.html:
```diff
@@ -0,0 +1,40 @@
+<h2>TV Shows</h2>
```
You probably forgot to update the header here!

gabriel-detassigny commented on app/views/addons/showaddon.html:
```diff
@@ -0,0 +1,27 @@
+  <li><a ng-href="#/tvshows">TV Shows</a></li>
+  <li><a ng-href="#/tvshows/{{ season.tvshowId }}">{{ season.tvshow }}</a></li>
+  <li class="active">Season {{ season.season }}</li>
+</ol>
```
Same here, the header and breadcrumb needs to be updated to reflect the Addons section

aborovsky replied:
Okay, i'll clean my code :)

gabriel-detassigny commented:
Thanks for the pull request! I like the idea of having an addons section, it's pretty cool. 😃 

Somehow the thumbnails don't show up for me though... Not sure why exactly 😕 

I get a 404 in the JS console, it's trying to load this URL : 
`http://0.0.0.0:9000/%7B%7B%20file.thumbnail%20|%20kodiThumbnailUrl%20%7D%7D`

aborovsky commented:
> I get a 404 in the JS console, it's trying to load this URL : 
> http://0.0.0.0:9000/%7B%7B%20file.thumbnail%20|%20kodiThumbnailUrl%20%7D%7D

I've fixed it.
Just replaced
`src="{{ file.thumbnail | kodiThumbnailUrl }}"`
with
`ng-src="{{ file.thumbnail | kodiThumbnailUrl }}"`

> Somehow the thumbnails don't show up for me though... Not sure why exactly :confused:

Hm, strange.
Thumbnails worked for me.
![image](https://cloud.githubusercontent.com/assets/1669420/17092893/555089ac-524e-11e6-8113-8b0f436cd37d.png)

Did you configure http port (ususally 8080) inside of `config.js`?

```
var configValues = {
  ...
  KODI_HTTP_PORT: 8080, // TCP port for HTTP connection to Kodi (acquire thumbnails)
  ...
};
```

Take a look at new [config.js.example](https://github.com/estbeetoo/kodi-angular/blob/e00d07b469bc2ca2077d0fb66e7906213ce5699b/app/scripts/config.js.example)

aborovsky commented:
Added [Code clenup commit](https://github.com/gabriel-detassigny/kodi-angular/pull/2/commits/b079a69c5b2ac7cba7769feb84028f2645ab4d93).
Please, review.

gabriel-detassigny commented:
Yeah, my HTTP port is set in the config. But I see why now!
When trying on youtube, it tries to load this URL for me : 

> http://192.168.1.10:8080/image//home/osmc/.kodi/addons/plugin.video.youtube/resources/media/sign_in.png

As you can see, there seems to be an extra `/` after image. Removing it fixes it, so if you change that in your URL parsing, it should be fine

gabriel-detassigny commented:
Ah and another thing : if you type `grunt jshint:all` on your branch, there is a warning. It's just some missing `{`, so should be very quick to fix :) thanks!

aborovsky commented:
All fixed, please review :)

gabriel-detassigny commented on app/scripts/services/websocket.js:
```diff
@@ -80,8 +80,30 @@ services.service('KodiWS', ['$q', '$window', 'KODI_URL', 'KODI_PORT', 'SOCKET_TI
+    var tmpThumbUrl = decodeURIComponent(thumbUrl);
+    if (tmpThumbUrl.indexOf('http://') === 0 || tmpThumbUrl.indexOf('https://') === 0)
+      return tmpThumbUrl;
+    return 'http://' + $btp.getCurrent().url + ':' + $btp.getCurrent().http_port + '/image/' + thumbUrl;
```
This part gives me this error : `ReferenceError: $btp is not defined`. 
What's $btp here? 
Shouldn't KODI_URL and KODI_HTTP_PORT do the job?

aborovsky replied:
Fixed!

gabriel-detassigny commented on app/scripts/services/websocket.js:
```diff
@@ -80,8 +80,30 @@ services.service('KodiWS', ['$q', '$window', 'KODI_URL', 'KODI_PORT', 'SOCKET_TI
+    if (thumbUrl.charAt(thumbUrl.length - 1) === '/') {
+      thumbUrl = thumbUrl.substring(0, thumbUrl.length - 1);
+    }
+    if (thumbUrl.charAt(0) === '/') {
```
That doesn't seem to remove it actually. I think it's because it is URL encoded at this point, so it's never a `/` here. Maybe do this after decodeURIComponent?

aborovsky replied:
Fixed too!

aborovsky commented:
jshint warns fixed, please review PR

aborovsky commented:
Good news! :)
пн, 10 окт. 2016 г. в 22:01, Gabriel de Tassigny notifications@github.com:

> Merged #2 https://github.com/gabriel-detassigny/kodi-angular/pull/2.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/gabriel-detassigny/kodi-angular/pull/2#event-818475397,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ABl5LHye3jvWoM2stdmsUZc--H2HGqIXks5qyotxgaJpZM4JRqR0
> .
> 
> ## 
> 
> [image: BeeToo] http://beetoo.me/
> Когда умный дом по-настоящему умный
> Александр Боровский, Технический директор
> Alexander Borovsky, CTO, Co-Founder
> +7-926-0-945-00 <+7926094500>
> +7-499-391-33-62
> <+74993913362>aborovsky@beetoo.me
> http://beetoo.me
